adding ipfs thing 2023 recap post (#67)
libp2p/blog: src/_blog/libp2p-IPFSþing-2023-recap.md

---
tags:
- libp2p
title: libp2p at IPFS þing 2023 Recap
description: Recap of libp2p presentations and workshops at IPFS þing 2023

> date: 2023-05-05

Reviewer: ```suggestion
date: 2023-05-11
```

> permalink: "/2023-05-05-libp2p-IPFS%C3%BEing-recap/"

Reviewer: ```suggestion
permalink: "/2023-libp2p-IPFS%C3%BEing-recap/"
```

translationKey: ''
header_image: /ipfs-thing-2023-logo.png
author: Dave Huseby
---

**Table of Contents**

[[toc]]

## Introduction

> Last month, April 15th - 19th 2023, the IPFS community came together in Brussels, Belgium for [IPFS þing 2023](https://blog.ipfs.tech/2023-ipfs-thing-recap/). The libp2p users and contributors community came out to meet up once again to interface with the broader IFPS community as well as share all of the great accomplishments and new work going on in the libp2p project. It was great to reconnect with people from organizations supporting and contributing to libp2p such as [Peergos](https://peergos.org/), [Source Network](https://source.network/), 

Reviewer 2: Not sure if there was a third group to call out at the end of the sentence here?

```suggestion
Last month, April 15th - 19th 2023, the IPFS community came together in Brussels, Belgium for [IPFS þing 2023](https://blog.ipfs.tech/2023-ipfs-thing-recap/). The libp2p users and contributors community came out to meet up once again to interface with the broader IFPS community as well as share all of the great accomplishments and new work going on in the libp2p project. It was great to reconnect with people from organizations supporting and contributing to libp2p such as [Peergos](https://peergos.org/) and [Source Network](https://source.network/).
```

<div class="container" style="display:flex; column-gap:10px;">
    <figure>
        <img src="../assets/ipfs-thing-2023-03.jpg"  width="100%">
        <figcaption style="font-size:x-small;">Group shot of the IPFSþing 2023</a>
        </figcaption>
    </figure>
</div>

Over the course of five days the the libp2p community gave 6 different talks on the recent libp2p developments and finished strong with a workshop where participants built their own peer-to-peer chat application leveraging the same technology as the [Universal Connectivity](https://github.com/libp2p/universal-connectivity) demonstrator project.

## Goals

> The goals of the libp2p contributors attending IPFSþing were:

Reviewer: ```suggestion
The goals of the libp2p contributors attending IPFS þing were:
```

1. Build excitement by demonstrating the [Universal Connectivity application](https://github.com/libp2p/universal-connectivity).
2. Give updates on the continually improving performance, dealing with non-uniform network topology, interoperability improvements, and lowering the barriers to libp2p compatibility.

> 3. Reconnect with community contributors and building up the greater libp2p community.

Reviewer: ```suggestion
3. Reconnect with community contributors and build up the greater libp2p community.
```

## Recap of Talks

@[youtube](4v-iIB0C9_8)

[Max Inden](https://github.com/mxinden) (rust-libp2p maintainer, Software Engineer at Protocol Labs)

> Max's talk about the [Universal Connectivity demonstrator app](https://github.com/libp2p/universal-connectivity) broke the record for the most people involved: [@2color / Daniel](https://github.com/2color), [@TheDiscordian](https://github.com/TheDiscordian), [@jochasinga / Pan](https://github.com/jochasinga), [@achingbrain / Alex](https://github.com/achingbrain), [@maschad / Chad](https://github.com/maschad), [@p-shahi / Prithvi](https://github.com/p-shahi), [@marcopolo / Marco](https://github.com/marcopolo), [@thomaseizinger / Thomas](https://github.com/thomaseizinger), and [@mxinden / Max](https://github.com/mxinden) all collaborated to make it happen. The work on the Universal Connectivity demonstrator app turned out to be an excellent driver for [interop testing and the need for further work there](https://github.com/libp2p/test-plans). The live demo showed a go-libp2p server talking to a rust-libp2p server talking to go-libp2p laptop talking to js-libp2p browsers using a variety of transports including [QUIC](https://github.com/libp2p/specs/tree/master/quic), [WebRTC Direct](https://github.com/libp2p/specs/blob/master/webrtc/webrtc-direct.md), [WebRTC](https://github.com/libp2p/specs/blob/master/webrtc/webrtc.md), and [WebTransport](https://github.com/libp2p/specs/tree/master/webtransport).

Reviewer 3: ```suggestion
Max's talk about the [Universal Connectivity demonstrator app](https://github.com/libp2p/universal-connectivity) broke the record for the most people involved: [@2color / Daniel](https://github.com/2color), [@TheDiscordian](https://github.com/TheDiscordian), [@jochasinga / Pan](https://github.com/jochasinga), [@achingbrain / Alex](https://github.com/achingbrain), [@maschad / Chad](https://github.com/maschad), [@p-shahi / Prithvi](https://github.com/p-shahi), [@marcopolo / Marco](https://github.com/marcopolo), [@thomaseizinger / Thomas](https://github.com/thomaseizinger), and [@mxinden / Max](https://github.com/mxinden) all collaborated to make it happen. The live demo showed a go-libp2p server talking to a rust-libp2p server talking to go-libp2p laptop talking to js-libp2p browsers using a variety of transports including [QUIC](https://github.com/libp2p/specs/tree/master/quic), [WebRTC Direct](https://github.com/libp2p/specs/blob/master/webrtc/webrtc-direct.md), [WebRTC](https://github.com/libp2p/specs/blob/master/webrtc/webrtc.md), and [WebTransport](https://github.com/libp2p/specs/tree/master/webtransport). The work on the Universal Connectivity demonstrator app turned out to be an excellent driver for [interop testing and the need for further work there](https://github.com/libp2p/test-plans).
```

Nit pick, suggest to reorder the last two sentences. The fact that it uses many transports is more interesting to a reader.

@[youtube](2h9jth3nvJw)
[Max Inden](https://github.com/mxinden) (rust-libp2p maintainer, Software Engineer at Protocol Labs)<br>
[Marco Munizaga](https://github.com/marcopolo) (go-libp2p and zig-libp2p maintainer, Software Engineer at Protocol Labs)

Watch Max and Marco describe how libp2p maintainers think about and measure performance, plus learn about some of the optimizations that are in the latest versions. This talk captures the in-progress work by libp2p maintainers to have automated, realistic, and easily reproducible benchmarks for the implementations.

@[youtube](Ixyo1G2tJZE)
[Marten Seemann](https://github.com/marten-seemann) (go-libp2p maintainer, Software Engineer at Protocol Labs)<br>
[Marco Munizaga](https://github.com/marcopolo) (go-libp2p and zig-libp2p maintainer, Software Engineer at Protocol Labs)

> Marten, with the help of Marco, presented on advancements in using HTTP(S) over libp2p. HTTP(S) is a universally supported protocol with lots of established infrastructure but it has some key limitations that libp2p solves. In this talk Marco and Marten demonstrate the use of js-libp2p and service workers to intercept normal HTTP(S) calls in the browser and re-route them over the libp2p connection to a peer that handles the HTTP(S) request and serves the content back over the libp2p connection. This approach solves the problem of making web apps able to use both normal HTTP(S) to access regular web servers and libp2p connections to access HTTP(S) servers running as libp2p peers.

Reviewer: ```suggestion
Marten, with the help of Marco, presented on advancements in using [HTTP(S) over libp2p](https://github.com/libp2p/specs/pull/508). HTTP(S) is a universally supported protocol with lots of established infrastructure but it has some key limitations that libp2p solves. In this talk Marco and Marten demonstrate the use of js-libp2p and service workers to intercept normal HTTP(S) calls in the browser and re-route them over the libp2p connection to a peer that handles the HTTP(S) request and serves the content back over the libp2p connection. This approach solves the problem of making web apps able to use both normal HTTP(S) to access regular web servers and libp2p connections to access HTTP(S) servers running as libp2p peers.
```

@[youtube](aDHymXQJ4bs)
[Marten Seemann](https://github.com/marten-seemann) (go-libp2p maintainer, Software Engineer at Protocol Labs)<br>
[Marco Munizaga](https://github.com/marcopolo) (go-libp2p and zig-libp2p maintainer, Software Engineer at Protocol Labs)

> Don't miss this presentation by Marten and Marco. They start off showing how libp2p is evolving and improving with the addition of the QUIC transport followed by an explanation of how simple it is to create a compatible libp2p stack out of a QUIC library, the libp2p TLS extension and some code for doing peer ID encoding and sending multistream headers. To prove how simple it is, Marco showed of his zig-libp2p implementation sending a ping to a go-libp2p node.

Reviewer: ```suggestion
Don't miss this presentation by Marten and Marco. They start off showing how libp2p is evolving and improving with the addition of the QUIC transport followed by an explanation of how simple it is to create a compatible libp2p stack out of a QUIC library, the libp2p TLS extension and some code for doing peer ID encoding and sending multistream headers. To prove how simple it is, Marco showed off his zig-libp2p implementation sending a ping to a go-libp2p node.
```

---
tags:
- libp2p

> title: libp2p at IPFSþing 2023 Recap

Reviewer: ```suggestion
title: libp2p at IPFS þing 2023 Recap
```
space between IPFS and þing

> description: Recap of libp2p presentations and workshops at IPFSþing 2023

Reviewer: ```suggestion
description: Recap of libp2p presentations and workshops at IPFS þing 2023
```

date: 2023-05-05
permalink: "/2023-05-05-libp2p-IPFS%C3%BEing-recap/"
translationKey: ''

> header_image: /ipfs-thing-2023-logo.svg

Reviewer: For some reason the header image is cutoff in the preview here https://bafybeido4ueaawkfbfw5h5dmd3327kumgjdupl5qmd3jc7nrybdnpem6wu.on.fleek.co/

looks fine in the actual article though

Author: It's because the image is an .svg. I'll render it to a .png and update it.

Author: fixed in the new version I'm about to push

author: Dave Huseby
---

**Table of Contents**

[[toc]]

## Introduction

> Last month, April 15th - 19th 2023, the IPFS community came together in Brussels, Beligium for [IPFSþing 2023](https://blog.ipfs.tech/2023-ipfs-thing-recap/). The libp2p users and contributors community came out to meet up once again to interface with the broader IFPS community as well as share all of the great accomplishements and new work going on in the libp2p project. It was great to reconnect with people from organizations supporting and contributing to libp2p such as [Peergos](https://peergos.org/), [Source Network](https://source.network/), 

Reviewer: ```suggestion
Last month, April 15th - 19th 2023, the IPFS community came together in Brussels, Belgium for [IPFS þing 2023](https://blog.ipfs.tech/2023-ipfs-thing-recap/). The libp2p users and contributors community came out to meet up once again to interface with the broader IFPS community as well as share all of the great accomplishments and new work going on in the libp2p project. It was great to reconnect with people from organizations supporting and contributing to libp2p such as [Peergos](https://peergos.org/), [Source Network](https://source.network/), 
```

incomplete sentence after "Source Network,"

Reviewer 4: I unresolved this because it still is an incomplete sentence when looking at https://bafybeibhflsivctriduyzeagxmueqzctyce4l75qibeudrdgmuqhu7x4ye.on.fleek.co/2023-05-05-libp2p-IPFS%C3%BEing-recap/

Reviewer: I pushed a fix

<div class="container" style="display:flex; column-gap:10px;">
    <figure>
        <img src="../assets/ipfs-thing-2023-03.jpg"  width="500">
        <figcaption style="font-size:x-small;">Group shot of the IPFSþing 2023</a>
        </figcaption>
    </figure>
</div>

Over the course of five days the the libp2p community gave 4 different talks on the recent libp2p developments and finished strong with a workshop where participants built their own peer-to-peer chat application leveraging the same technology as the [Universal Connectivity](https://github.com/libp2p/universal-connectivity) demonstrator project.

## Goals

The goals of the libp2p contributors attending IPFSþing were:

1. Build excitement by demonstrating the [Universal Connectivity application](https://github.com/libp2p/universal-connectivity).

> 2. Give updates on the continually improving performace, dealing with non-uniform network topoligy with hole punching, and lowering the barriers to libp2p compatibility.

Reviewer: ```suggestion
2. Give updates on the continually improving performance, dealing with non-uniform network topology with hole punching, and lowering the barriers to libp2p compatibility.
```

Reviewer 5: I don't think we should mention hole punching here. That was something we worked on a long time in the past, let's talk about the new stuff!

Author: I pulled out the mention of hole punching.

3. Reconnect with community contributors and building up the greater libp2p community.

## Recap of Talks

### Hole Punching in the Wild

> Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)

Author: the new version doesn't have the titles so we can see how it looks.

> Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)

Reviewer 4: I still personally think the names/titles under the videos add noise and would remove them.  But if we want to keep them, I think we should do a couple of things:

1. Make them look more like a caption directly under the video.  Maybe smaller text, italic, less space between video and text?
2. Add some sort of a separator (e.g., light horizontal line, bit more space) between each video?  See screenshot below for where the spacing seems doesn't make clear that it's a "section" break (new video):
<img width="1060" alt="image" src="https://github.com/libp2p/blog/assets/85411/a2d71a1d-7200-4404-a5f8-79da4f1751bb">

> Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)

Reviewer 4: Actually, for delinating videos, how about adding a header like we have for "libp2p workshop".  That also gives the nice property of being able to anchor link to a section.  Each heading can be the talk name (and optionally the speakers).

> Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)

Reviewer: Is there a reason to diverge from this https://blog.libp2p.io/2022-11-22-libp2p-day-2022-recap/#recap-of-talks

I think that's how @dhuseby originally had it. Can we do that please?

> Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)

Author: @BigLep goo point on the anchor linking to sections. I put titles in again to delineate.

@[youtube](R-ToBsdlEk4)

> Learn from Max on how libp2p's hole punching mechanism works and what we've learned from the measurement campaign run last year.

Author: added

### libp2p Performance

> <div class="container" style="display:flex; column-gap:10px;">

Author: I agree...updated version doesn't have these

> <div class="container" style="display:flex; column-gap:10px;">

Author: I move the photos to the bottom of the page in the gallery section to remove the clutter.

    <figure>
    <figure>
        <img src="../assets/ipfs-thing-2023-02.jpg" width="100%">
        <figcaption style="font-size:x-small;">Max Inden and Marco Munizaga entertain us with a skit about libp2p performance.
            <a href="https://app.iconik.io/share/assets/16fdde28-dd2c-11ed-9cf1-fe682d265912/?object_type=collections&object_id=3a266a5e-dd28-11ed-9faf-e667e71f3893&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzaGFyZV9pZCI6IjBkM2VhZDJjLWRkMzEtMTFlZC1iYzA3LTFhMGQ4ZDEzMTExNCIsInNoYXJlX3VzZXJfaWQiOiIwZDRmMjQwZS1kZDMxLTExZWQtYmMwNy0xYTBkOGQxMzExMTQiLCJleHAiOjE4Mzk0MjM4MjZ9.OjJFG2ItCyURSqwzvxmh6dyFVo7LHwdR2puBq0KAm00">[Pic credit.]</a>
        </figcaption>
    </figure>
</div>

Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)
Marco Munizaga (go-libp2p and zig-libp2p maintainer, Software Engineer at Protocol Labs)

@[youtube](2h9jth3nvJw)

Watch Max and Marco talk about all of the performance related aspects of libp2p and what optimizations are in the latest versions. They give a great overview of how the Protocol Labs libp2p team does automated performance testing using terraform and two AWS servers on opposite ends of the United States.

> ### The Incredible Benefits of libp2p + HTTP: A Match Made in Decentralized Heaven

Reviewer 4: The text/video below doesn't match this title.

That's a separate talk, so it looks like we're missing an entry maybe?

Author: fixed

<div class="container" style="display:flex; column-gap:10px;">
    <figure>
        <img src="../assets/ipfs-thing-2023-01.jpg"  width="100%">
        <figcaption style="font-size:x-small;">Marco Munizaga and Marten Seemann explain how to build your own compatible libp2p stack.
            <a href="https://app.iconik.io/share/assets/8b123fe8-dd2c-11ed-8154-3e3ce49f182b/?object_type=collections&object_id=3a266a5e-dd28-11ed-9faf-e667e71f3893&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzaGFyZV9pZCI6IjBkM2VhZDJjLWRkMzEtMTFlZC1iYzA3LTFhMGQ4ZDEzMTExNCIsInNoYXJlX3VzZXJfaWQiOiIwZDRmMjQwZS1kZDMxLTExZWQtYmMwNy0xYTBkOGQxMzExMTQiLCJleHAiOjE4Mzk0MjM4MjZ9.OjJFG2ItCyURSqwzvxmh6dyFVo7LHwdR2puBq0KAm00">[Pic credit.]</a>
        </figcaption>
    </figure>
</div>

Marten Seemann (go-libp2p maintainer, Software Engineer at Protocol Labs)
Marco Munizaga (go-libp2p and zig-libp2p maintainer, Software Engineer at Protocol Labs)

@[youtube](aDHymXQJ4bs)

> Don't miss this presentation by Marten and Marco. They start off showing how libp2p is evolving and improving with the addition of the QUIC transport followed by an explanation of simple it is to create a compatible libp2p stack out of a QUIC library, the libp2p TLS extension and some code for doing peer ID encoding and sending multistream headers. To prove how simple it is, Marco showed of his zig-libp2p implementation sending a pig to a go-libp2p node.

Reviewer: ```suggestion
Don't miss this presentation by Marten and Marco. They start off showing how libp2p is evolving and improving with the addition of the QUIC transport followed by an explanation of how simple it is to create a compatible libp2p stack out of a QUIC library, the libp2p TLS extension and some code for doing peer ID encoding and sending multistream headers. To prove how simple it is, Marco showed of his zig-libp2p implementation sending a ping to a go-libp2p node.
```
🐷 I wish pings were called pigs 😆

### Connecting everthing, everywhere, all at once with libp2p

Max Inden (rust-libp2p maintainer, Software Engineer at Protocol Labs)

@[youtube](4v-iIB0C9_8)

> Max's talk about the [Universal Connectivity demonstrator app](https://github.com/libp2p/universal-connectivity) broke the record for the most people involved. @2color / Daniel, @TheDiscordian, @jochasinga / Pan, @achingbrain / Alex, @maschad / Chad, @p-shahi / Prithvi, @marcopolo / Marco, @thomaseizinger / Thomas, and @mxinden / Max all collaborated to make it happen. The work on the Universal Connectivity demonstrator app turned out to be an excellent driver for [interop testing and the need for further work there](https://github.com/libp2p/interop). The live demo worked in showing a go-libp2p server talking to a rust-libp2p server talking to go-libp2p laptop talking to js-libp2p browsers all using a variety of transports including QUIC, WebRTC Direct, and WebTransport. Max also introduced us to the [new libp2p website focused on connectivity](https://connectivity.libp2p.io/).

Reviewer: ```suggestion
Max's talk about the [Universal Connectivity demonstrator app](https://github.com/libp2p/universal-connectivity) broke the record for the most people involved. @2color / Daniel, @TheDiscordian, @jochasinga / Pan, @achingbrain / Alex, @maschad / Chad, @p-shahi / Prithvi, @marcopolo / Marco, @thomaseizinger / Thomas, and @mxinden / Max all collaborated to make it happen. The work on the Universal Connectivity demonstrator app turned out to be an excellent driver for [interop testing and the need for further work there](https://github.com/libp2p/test-plans). The live demo worked in showing a go-libp2p server talking to a rust-libp2p server talking to go-libp2p laptop talking to js-libp2p browsers all using a variety of transports including [QUIC](https://github.com/libp2p/specs/tree/master/quic), [WebRTC Direct](https://github.com/libp2p/specs/blob/master/webrtc/webrtc-direct.md), [WebRTC](https://github.com/libp2p/specs/blob/master/webrtc/webrtc.md), and [WebTransport](https://github.com/libp2p/specs/tree/master/webtransport).
```

### libp2p Workshop

Thomas Eizinger (rust-libp2p maintainer, Software Engineer at Protocol Labs)

> During the unconference sessions late in the week, Thomas hosted a libp2p workshop where participants went through the process of building a peer-to-peer chat application using the rust-libp2p. The workshop's code is in [this repo](https://github.com/thomaseizinger/libp2p-workshop.git). You too can give it a try by cloning the code with `--mirror` so that you get all of the branches. This workshop is designed to be followed in iterations. Start by checking out the `iteration-1` branch and follow along in the README.md. Once complete, you can check out the `iteration-2` branch and again follow along the additional steps in the READM.md. The repo contains branches for `iteration-1`, `iteration-2`, `iteration-3`, `iteration-4`, `iteration-5`, and `iteration-final`. When done you should have a functioning peer-to-peer chat application built using rust-libp2p.

Reviewer 4: ```suggestion
During the unconference sessions late in the week, Thomas hosted a libp2p workshop where participants went through the process of building a peer-to-peer chat application using rust-libp2p. The workshop's code is in [this repo](https://github.com/thomaseizinger/libp2p-workshop.git). You too can give it a try by cloning the code with `--mirror` so that you get all of the branches. This workshop is designed to be followed in iterations. Start by checking out the `iteration-1` branch and follow along in the README. Once complete, you can check out the `iteration-2` branch and again follow along the additional steps in the README. The repo contains branches for `iteration-1`, `iteration-2`, `iteration-3`, `iteration-4`, `iteration-5`, and `iteration-final`. When done you should have a functioning peer-to-peer chat application built using rust-libp2p.
```

> During the unconference sessions late in the week, Thomas hosted a libp2p workshop where participants went through the process of building a peer-to-peer chat application using the rust-libp2p. The workshop's code is in [this repo](https://github.com/thomaseizinger/libp2p-workshop.git). You too can give it a try by cloning the code with `--mirror` so that you get all of the branches. This workshop is designed to be followed in iterations. Start by checking out the `iteration-1` branch and follow along in the README.md. Once complete, you can check out the `iteration-2` branch and again follow along the additional steps in the READM.md. The repo contains branches for `iteration-1`, `iteration-2`, `iteration-3`, `iteration-4`, `iteration-5`, and `iteration-final`. When done you should have a functioning peer-to-peer chat application built using rust-libp2p.

Reviewer 5: Does it make sense to mention this, given that it was not recorded?

Author: I think it is valuable because it is content that people can follow along with. The call to action is "you too can give it a try..."

We wanted to thank all members of the libp2p community including the maintainers, contributors, and supports of the work. Excitement is growing in libp2p and there are many opportunities for new people to get involved in the future of the project:

- If you’d like to get involved and contribute to libp2p, you can reach out to us using these means: [https://libp2p.io/#community](https://libp2p.io/#community)
- If you’re a self-starter and want to start pushing code immediately, feel free to ping the maintainers in any of these help wanted/good first issues: [go-libp2p](https://github.com/libp2p/go-libp2p/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), [js-libp2p](https://github.com/libp2p/js-libp2p/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), and [rust-libp2p](https://github.com/libp2p/rust-libp2p/issues?q=is%3Aopen+is%3Aissue+label%3Agetting-started).
- If you want to work in and around libp2p full-time, there are various teams hiring including the implementation teams.  See [https://jobs.protocol.ai/jobs?q=libp2p](https://jobs.protocol.ai/jobs?q=libp2p) for opportunities across the [Protocol Labs Network](https://plnetwork.io/).

> - There is a call for participants in the [NAT Hole Punching Measurement Campaign](https://discuss.libp2p.io/t/call-for-participation-nat-hole-punching-measurement-campaign/1690). Please check it out, any help is appreciated!

Reviewer: ```suggestion
```
Let's remove this since the campaign's ended
